lib/Sema/TypeCheckProtocol.cpp

```diff
@@ -594,17 +594,23 @@ swift::matchWitness(
   } else if (isa<ConstructorDecl>(witness)) {
     decomposeFunctionType = true;
     ignoreReturnType = true;
-  } else if (isa<EnumElementDecl>(witness)) {
-    auto enumCase = cast<EnumElementDecl>(witness);
+  } else if (auto *enumCase = dyn_cast<EnumElementDecl>(witness)) {
+    // An enum case with associated values can satisfy only a
+    // method requirement.
     if (enumCase->hasAssociatedValues() && isa<VarDecl>(req))
       return RequirementMatch(witness, MatchKind::EnumCaseWithAssociatedValues);
-    auto isValid = isa<VarDecl>(req) || isa<FuncDecl>(req);
-    if (!isValid)
+
+    // An enum case can satisfy only a method or property requirement.
+    if (!isa<VarDecl>(req) && !isa<FuncDecl>(req))
       return RequirementMatch(witness, MatchKind::KindConflict);
-    if (!cast<ValueDecl>(req)->isStatic())
+
+    // An enum case can satisfy only a static requirement.
+    if (!req->isStatic())
       return RequirementMatch(witness, MatchKind::StaticNonStaticConflict);
+
+    // An enum case cannot satisfy a settable property requirement.
     if (isa<VarDecl>(req) &&
-        cast<VarDecl>(req)->getParsedAccessor(AccessorKind::Set))
+        cast<VarDecl>(req)->isSettable(req->getDeclContext()))
```

Comment:
A 'static let' would return false for isSettable(), but it doesn't have a parsed getter either. Maybe it's better to just check for both the presence of a parsed getter and the absence of a setter?

Reply:
Either that, or allow the conformance to succeed in this case, and fix the crash elsewhere. After all, if the protocol requirement is misdeclared, diagnosing the conforming type seems like a case of an unnecessary diagnostics cascade.

Reply:
Turns out a static let returns true for isSettable if I pass the decl context, because isSettable does not properly handle static properties. On the other hand, it is also weird to be asking whether a property requirement is settable in some arbitrary context. 

Why do you suggest to check for a *parsed* getter? It seems alright for the conformance to succeed despite the misdeclared 'static let' with no parsed accessors. The problem with 'static var' in the same scenario is how the  (probably) parser decides for it to be mutable. Since stuff like this works:

```swift
protocol P {
  static var foo: Int // error
}

func test(_ x: P.Type) {
   x.foo = 0 // OK
}
```

I think the conformance should also fail . WDYT?